src/app.ts

```diff
@@ -86,6 +109,132 @@ const cachingMiddleware = apicache.options({
+    .mapNullable(collection => {
+      return tryCatch(() => {
+        return Array.from(collection).filter(
+          elem => elem.getAttribute("Name") === "fiscalNumber"
+        )[0];
+      });
+    })
+    .mapNullable(maybeElem => maybeElem.toNullable())
+    .mapNullable(_ => _.textContent?.trim());
+};
+
+const getRequestIDFromPayload = (tagName: string, attrName: string) => (
+  doc: Document
+): Option<string> => {
+  return fromNullable(
+    doc.getElementsByTagNameNS(SAML_NAMESPACE.PROTOCOL, tagName).item(0)
+  ).chain(element =>
+    fromEither(NonEmptyString.decode(element.getAttribute(attrName)))
+  );
+};
+
+const getRequestIDFromRequest = getRequestIDFromPayload("AuthnRequest", "ID");
+const getRequestIDFromResponse = getRequestIDFromPayload(
+  "Response",
+  "InResponseTo"
+);
+
+const SpidBaseMsg = t.interface({
+  // Timestamp of Request/Response creation
+  createdAt: UTCISODateFromString,
+
+  // Date of the SPID request / response in YYYY-MM-DD format
+  createdAtDay: PatternString("^[0-9]{4}-[0-9]{2}-[0-9]{2}$"),
+
+  // IP of the client that made a SPID login action
+  ip: IPString,
+
+  // XML payload of the SPID Request/Response
+  payload: t.string,
+
+  // Payload type: REQUEST or RESPONSE
+  payloadType: t.keyof({ REQUEST: null, RESPONSE: null })
+});
+
+export type SpidBaseMsg = t.TypeOf<typeof SpidBaseMsg>;
+
+const callback = (
+  sourceIp: string | null,
+  payload: string,
+  payloadType: "REQUEST" | "RESPONSE"
+): void => {
+  const maybeXmlPayload = tryCatch(() =>
+    new DOMParser().parseFromString(payload, "text/xml")
+  );
+  if (isNone(maybeXmlPayload)) {
+    log.error(`SpidLogCallback|ERROR=Cannot parse SPID XML Payload`);
+    return;
+  }
+
+  const doc = maybeXmlPayload.value;
+  const requestId =
+    payloadType === "REQUEST"
+      ? getRequestIDFromRequest(doc)
+      : getRequestIDFromResponse(doc);
+  const tFiscalCode = getFiscalNumberFromPayload(doc);
+
+  if (requestId.isNone()) {
+    log.error(
+      `SpidLogCallback| ERROR | Cannot recognize Request ID on XML Payload`
+    );
+    return;
+  }
+
+  if (tFiscalCode.isNone() && payloadType === "RESPONSE") {
+    log.error(
+      `SpidLogCallback| ERROR | Cannot recognize fiscal Code on XML Payload provided by SAMLResponse`
+    );
+    return;
+  }
+  const item = {
+    fiscalCode: tFiscalCode.toUndefined(),
+    spidRequestId: requestId.toUndefined()
+  };
+
+  const maybeBaseMsg = SpidBaseMsg.decode({
+    createdAt: new Date(),
+    createdAtDay: dateFnsFormat(new Date(), "YYYY-MM-DD"),
+    ip: sourceIp as IPString,
+    payload,
+    payloadType
+  });
+
+  if (isLeft(maybeBaseMsg)) {
+    log.error(`SpidLogCallback| ERROR | Cannot decode SpiD base message`);
+    return;
+  }
+  const baseMsg = maybeBaseMsg.value;
+  const spidMessage = {
+    ...baseMsg,
+    ...item
+  };
+
+  spidQueueClient
+    .create()
+    .then()
+    .catch(_ => {
+      log.error(`SpidLogCallback| ERROR | Cannot recognize SpiD Queue`);
+    });
```

Comment:
```suggestion
```
we will create the queue though the infrastructure setup

Reply:
the `create()` method allows to create a new queue or subscribe to an existing one, so I think we should keep this code

Reply:
no we don't. moreover this call as-is runs in parallel with the following code so there's no grant that the queue gets created before sending the message to the queue. please remove this code, we'll create the queue within terraform